### ISSUE TITLE
Learning rate is in a line of hard code

### ISSUE THREAD

**Issue by PPWangyc** [paninski-lab/lightning-pose#267]
The learning rate in the litpose is set to 1e-3 in `lightning_pose/models/base.py`. It would be great to make [learning rate](https://github.com/paninski-lab/lightning-pose/blob/main/lightning_pose/models/base.py#L408C1-L425) can be specified in the config training file. 

Moreover, it would be nice to select different optimizer/scheduler. Thanks!

**Comment by ksikka** [comment]
Hi @PPWangyc thanks for your message. To customize learning rate, the easiest way is to subclass the classes and override `configure_optimizers`. Then patch the module to point to your class.

For example:

```
from lightning_pose.models import heatmap_tracker


class MyModel(HeatmapTracker):
  def configure_optimizers(self):
    pass # your code here ...


# Use MyModel instead of lightning-pose HeatmapTracker
heatmap_tracker.HeatmapTracker = MyModel
```

We will have to figure out how to get this code to run before lightning-pose creates the model object.
How are you running lightning-pose?

**Comment by themattinthehatt** [comment]
As part of this I think we can support a new `training.optimizer` field in the config file. This can default to `Adam` and have a configurable learning rate, and we can also include an `AdamW` with a learning rate and other hyperparameters

**Comment by PPWangyc** [comment]
Hi @ksikka , thanks so much for your information! I plan to use learning rate and optimizer to train my own model, so I need to sweep different parameters and settings in many runs. 

As you said, I can override the `configure_optimizers` function and insert to my own code. Thank you!

**Comment by ksikka** [comment]
Any time, by the way we are going to add support for learning rate in the config file to make it possible to do sweeps, and also allow you to choose between Adam and AdamW. Thanks!

**Comment by ksikka** [comment]
Hi @PPWangyc we added support for optimizer and learning rate, please see the linked PR. Feel free to open more issues as you need additional features! Thanks!

**Comment by PPWangyc** [comment]
Many thanks!
